### DISCUSSION TITLE
How to enable Front Channel Logout post_logout_redirection_uri to have spaces

### DISCUSSION THREAD

**Discussion by jonathan9li** [keycloak/keycloak#26817] [Q&A] [answered]
I want to have post_logout_redirec_uri to have a queryparam value with spaces
For example post_logout_redirect_uri=http://localhost:8100?param1=Hello how are you&idToken=[...]

Is it possible?

**Comment by danielFesenmeyer** [comment, marked as answer]
It should work, when you encode all your query params like this:

```
var param1Key = "param1";
var param1Value = "Hello how are you";

var queryString = URLEncoder.encode(param1Key, StandardCharsets.UTF_8) +
                "=" + URLEncoder.encode(param1Value, StandardCharsets.UTF_8);
```

The queryString should then contain: `...Hello+how+are+you `

When _post_logout_redirect_uri_ is part of a querystring itself (as part of a Keycloak logout uri), this should be encoded once more and will look like this: `...Hello%2Bhow%2Bare%2Byou`

Note the usage of the Java builtin URLEncoder instead of the Jakarta method UriBuilder#queryParam - the latter does not work well with such "nested querystrings".

**Reply by jonathan9li** [reply]
Hi Daniel, thank you for your quick response and explanation.
After encoding the queryString it works fine!
Thanks again!
